### PR TITLE
mgmt, local codegen for recoveryservicessiterecovery

### PR DESCRIPTION
**Pull request by XiaofeiCao**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/recoveryservicessiterecovery/azure-resourcemanager-recoveryservicessiterecovery/src/main/java/com/azure/resourcemanager/recoveryservicessiterecovery/fluent/StorageClassificationMappingsClient.java [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.core.http.rest.Response;
+import com.azure.core.management.polling.PollResult;
+import com.azure.core.util.Context;
+import com.azure.core.util.polling.SyncPoller;
+import com.azure.resourcemanager.recoveryservicessiterecovery.fluent.models.StorageClassificationMappingInner;
+import com.azure.resourcemanager.recoveryservicessiterecovery.models.StorageClassificationMappingInput;
+
+/** An instance of this class provides access to all the operations defined in StorageClassificationMappingsClient. */
+public interface StorageClassificationMappingsClient {
```
Comment (XiaofeiCao) [inline sdk/recoveryservicessiterecovery/azure-resourcemanager-recoveryservicessiterecovery/src/main/java/com/azure/resourcemanager/recoveryservicessiterecovery/fluent/StorageClassificationMappingsClient.java:18]: Renamed from `ReplicationStorageClassificationMappingsClient` to `StorageClassificationMappingsClient`.

File: sdk/recoveryservicessiterecovery/azure-resourcemanager-recoveryservicessiterecovery/src/main/java/com/azure/resourcemanager/recoveryservicessiterecovery/models/UpdateReplicationProtectedItemProviderSpecificInput.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    use = JsonTypeInfo.Id.NAME,
+    include = JsonTypeInfo.As.PROPERTY,
+    property = "instanceType",
+    defaultImpl = UpdateReplicationProtectedItemProviderSpecificInput.class)
+@JsonTypeName("UpdateApplianceForReplicationProtectedItemProviderSpecificInput")
+@JsonSubTypes({
+    @JsonSubTypes.Type(name = "InMageRcm", value = InMageRcmUpdateApplianceForReplicationProtectedItemInput.class)
+})
+@Immutable
+public class UpdateReplicationProtectedItemProviderSpecificInput {
```
Comment (XiaofeiCao) [inline sdk/recoveryservicessiterecovery/azure-resourcemanager-recoveryservicessiterecovery/src/main/java/com/azure/resourcemanager/recoveryservicessiterecovery/models/UpdateReplicationProtectedItemProviderSpecificInput.java:23]: Renamed from `UpdateApplianceForReplicationProtectedItemProviderSpecificInput` to `UpdateReplicationProtectedItemProviderSpecificInput`.

File: eng/mgmt/automation/generate.py [hunk 3]
```diff
@@ -231,7 +231,7 @@ def main():
     module = ARTIFACT_FORMAT.format(service)
     stable_version, current_version = set_or_increase_version(sdk_root, GROUP_ID, module, **args)
     args['version'] = current_version
-    output_folder = OUTPUT_FOLDER_FORMAT.format(service),
+    output_folder = OUTPUT_FOLDER_FORMAT.format(service)
```
Comment (XiaofeiCao) [inline eng/mgmt/automation/generate.py:234]: Trailing comma seems wrong.

File: sdk/recoveryservicessiterecovery/azure-resourcemanager-recoveryservicessiterecovery/src/main/java/com/azure/resourcemanager/recoveryservicessiterecovery/fluent/StorageClassificationMappingsClient.java (hunk 1 above)
Comment (weidongxu-microsoft) [reply sdk/recoveryservicessiterecovery/azure-resourcemanager-recoveryservicessiterecovery/src/main/java/com/azure/resourcemanager/recoveryservicessiterecovery/fluent/StorageClassificationMappingsClient.java:18]: Is it done manually or in codegen?

Comment (XiaofeiCao) [reply sdk/recoveryservicessiterecovery/azure-resourcemanager-recoveryservicessiterecovery/src/main/java/com/azure/resourcemanager/recoveryservicessiterecovery/fluent/StorageClassificationMappingsClient.java:18]: It's through local codegen.

autorest PR: https://github.com/Azure/autorest.java/pull/1911
specs PR: https://github.com/Azure/azure-rest-api-specs/pull/22040

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**